libcxx/cmake/caches/hexagon-linux-runtimes.cmake

```diff
@@ -0,0 +1,25 @@
+
+set(CMAKE_EXE_LINKER_FLAGS "-lclang_rt.builtins-hexagon -nostdlib" CACHE STRING "")
+set(CMAKE_SHARED_LINKER_FLAGS "-lclang_rt.builtins-hexagon -nostdlib" CACHE STRING "")
```

Comment:
Why are those flags necessary when building for Hexagon?

Reply:
I'll double check these too - I recall they were necessary during bootstrapping, but could have been due to driver bugs fixed since we added that.

Reply:
The first failure encountered when I remove these is related to `C_SUPPORTS_NODEFAULTLIBS_FLAG`.

And I think it's `llvm_check_compiler_linker_flag(C "-nodefaultlibs" C_SUPPORTS_NODEFAULTLIBS_FLAG)` from `llvm-project/libunwind/cmake/config-ix.cmake` failing.

Sorry if I'm being dense here but if the compiler feature test tries to build an executable using `-nodefaultlibs` but without providing the necessary libraries to satisfy a trivial C executable, wouldn't we expect it to fail?  I mean, I guess other compilers must be succeeding here, so I'll have to try and understand how.  Or maybe it could be that `-nodefaultlibs` is mishandled by the hexagon driver?  I guess I'd figure it would remove at least `-lc` though, which seems like it should fail something like this.

```
Change Dir: /local/mnt/workspace/upstream/toolchain_for_hexagon/obj_libs/CMakeFiles/CMakeScratch/TryCompile-z4nhCq

Run Build Command(s):/pkg/qct/software/ninja/1.8.2/ninja cmTC_11433 && [1/2] Building C object CMakeFiles/cmTC_11433.dir/src.c.o
[2/2] Linking C executable cmTC_11433
FAILED: cmTC_11433 
: && /local/mnt/workspace/upstream/toolchain_for_hexagon/clang+llvm-19.0.0-cross-hexagon-unknown-linux-musl/x86_64-linux-gnu/bin/hexagon-unknown-linux-musl-clang --target=hexagon-unknown-linux-musl -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections  --start-no-unused-arguments --unwindlib=none --end-no-unused-arguments -nodefaultlibs  CMakeFiles/cmTC_11433.dir/src.c.o -o cmTC_11433   && :
ld.lld: error: undefined symbol: __libc_start_main
>>> referenced by crt1.c
>>>               /local/mnt/workspace/upstream/toolchain_for_hexagon/clang+llvm-19.0.0-cross-hexagon-unknown-linux-musl/x86_64-linux-gnu/bin/../target/hexagon-unknown-linux-musl//usr/lib/crt1.o:(_start_c)
>>> referenced by crt1.c
>>>               /local/mnt/workspace/upstream/toolchain_for_hexagon/clang+llvm-19.0.0-cross-hexagon-unknown-linux-musl/x86_64-linux-gnu/bin/../target/hexagon-unknown-linux-musl//usr/lib/crt1.o:(_start_c)
hexagon-unknown-linux-musl-clang: error: hexagon-ld command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.


Source file was:
int main(void) { return 0; }

```

Reply:
Usually, `-nodefaultlibs` should work if you don't make any calls to standard library functions. In this case it looks like the compiler is relying on an external library to provide `__libc_start_main` (presumably the thing that calls `main`), which causes linker errors like this.

I think you want to try setting `CMAKE_TRY_COMPILE_TARGET_TYPE` to `STATIC_LIBRARY` instead.

Reply:
Thanks for the suggestion, it looks like this uncovers a driver bug.  I'll take a look at that first.

Reply:
Ok, I think the driver bug (`--unwindlib=none` was ignored) might not have been the critical factor.  I'll open a PR for that regardless.  But I made a few fixes to address these and perhaps some of the other concerns you raised.

Reply:
>  I think the driver bug (--unwindlib=none was ignored) might not have been the critical factor.

This was fixed in #99552.  And these flag overrides have now been removed.